cpp/include/cudf_test/column_utilities.hpp

```diff
@@ -216,6 +216,21 @@ inline std::pair<thrust::host_vector<std::string>, std::vector<bitmask_type>> to
   return {host_data, bitmask_to_host(c)};
 }
 
+/**
+ * @brief Stringify the inconsistent values resulted from the comparison of two columns element-wise
+ *
+ * @param differences stores the indices at which the given columns have different values
+ * @param lhs and rhs are the two input columns
+ * @param print_all_differences true: print all differing values; false: print only the first
+ * differing value
+ * @param depth the nested level of the differing values in the given columns
+ */
+std::string stringify_column_differences(thrust::device_vector<int> const& differences,
```

Comment:
Does this really need to be in the public header? I don't think there's a way to get a `differences` vector using public APIs.

Reply:
We generally don't need it to be public either. However, I still need to access it to run the test (`byte_cast_tests.cpp`). We can have an option to bring that function back to private, but then we have to remove the crash test and assume that this fix will work forever---the thrust crash will not come back.

Reply:
I don't see how assuming the fix works if the dependent tests pass is any different from any of our other code and tests. I don't think we need to try to invoke a crash in an internal function that wouldn't otherwise be explicitly tested.

Reply:
I agree. We typically test to see a computation result is correct or not---not very commonly test to see if a function would crash or not.